### PR TITLE
Add onelogin_brands resource for account branding

### LINKED ISSUE #263: Branding Resource
Can a resource for Branding API be brought back into the provider please. It was there at one point before with older versions

### PR DESCRIPTION
**Pull request by Subterrane**
Closes #263.

## The premise needed correcting first

The issue asks for the Branding resource to be "brought back". It was never there.

`onelogin_brands` has **never existed in provider Go code** — not in any commit, on any branch. What existed was *documentation*: `docs/resources/brands.md` plus four `docs/data-sources/brands*.md`, added March 2023 and deleted on 27 Mar 2023 in `eb79237` ("brand document removal"). The string `onelogin_brands` appears only in those docs, the README and `swag-api.yml`; it was never in `ResourcesMap`. So this is a new resource, not a restoration — but the reporter's belief was entirely reasonable, since published docs described a schema with nothing behind it.

The SDK side needed nothing: `onelogin-go-sdk` has carried full brand CRUD (`CreateBrand`, `GetBrandByID`, `UpdateBrand`, `DeleteBrand`, `ListAccountBrands`) for some time.

## Design decisions, each checked against the API

I probed the Development tenant rather than inferring behaviour from the model:

| Decision | Evidence |
|---|---|
| `name` is Required | create with `{}` → `422 {"name":"Value is required."}` |
| `name` always sent on update | `{"name":null,...}` → **422** "Value must be a string."; omitting the key → **200** |
| `logo`/`background` write-only | sent as base64, returned as `{file_size, content_type, updated_at, urls{navigation,login,original}}` |
| localised fields compare decoded | API stores `custom_label_text_for_login_screen` etc. as a JSON locale map encoded in a string, and re-encodes it |
| everything else Optional+Computed | create fills defaults: `enabled` false, `custom_color` `#000000`, `custom_accent_color` `#016B91`, `custom_masking_color` `#000000`, `custom_masking_opacity` 0 |

### A latent SDK bug this steers around

`models.Brand.Name` is `json:"name"` with **no `omitempty`** — byte-for-byte the defect fixed on `models.App.Name` in SDK v4.17.0, still present on `Brand`. A partial update built with a nil `Name` would send `"name": null` and be refused outright.

`RequestBody` always sets `Name`, so this resource never produces that request whatever the SDK does. I'll raise the `omitempty` fix upstream separately; **this PR does not depend on it.**

### Deliberately not managed

The API returns `custom_links`, `use_custom_smtp_setting`, and on the master brand `domain_name`, `allowed_redirect_urls`, `hide_login_forgot_password` and the `password_self_service_*` fields. None is on `models.Brand`, so none can be written through the SDK. They are left out and documented rather than offered as attributes that would be silently dropped.

## Verification

**End to end against the Development tenant**, with a `dev_overrides` build of this branch — not just unit tests:

- `apply` → created, `master = false`
- **second `plan` → `No changes`** (the real test of Optional+Computed)
- reordered the locale keys and reformatted the JSON → **still `No changes`** (DiffSuppressFunc working on live data)
- `apply` with a changed colour → modified, then `No changes`
- `state rm` + `import` → `Import successful!`, then `No changes`
- deleted the brand out of band → plan offers **`1 to add`** rather than erroring (the `IsNotFoundError` path)
- `destroy` → destroyed

The tenant was left exactly as found — the probe brands were deleted and the four pre-existing brands are untouched.

**Unit tests** (`ol_schema/brand`): schema shape, `ConfiguredKeys`, `RequestBody`, and seven `DiffSuppressFunc` cases including malformed input on either side.

Two assertions were mutation-checked rather than assumed:
- removing a `case` from the `assign` switch fails `TestRequestBodyCoversEveryField` by name — that test exists because a field added to the table but not the switch would be accepted in a configuration and then silently never sent
- making `Name` conditional fails `TestRequestBody/always_sends_the_name`

`go build`, `go vet` and `gofmt` clean. `go test ./...` passes except `TestAccPreCheck`, which fails identically on `main` (`ONELOGIN_CLIENT_ID must be set`).

### CODE REVIEW

File: ol_schema/brand/brand.go [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+		// against the API.
+		//
+		// The cost is that a logo replaced outside Terraform is not detected
+		// as drift: state keeps the base64 that was last applied. Changing the
+		// configured value still replaces the image.
+		"logo": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Base64-encoded PNG for the login page logo, under 1MB. Write-only: the API returns image URLs rather than the data, so a logo changed outside Terraform is not seen as drift.",
+		},
```
[the remark this answers is not in the document]
Comment (Subterrane) [reply ol_schema/brand/brand.go:105]: Applied in fea8d32 — both `logo` and `background` are now `Sensitive`.

One correction to the reasoning, since it matters for anyone reading this later: the precedent cited does not quite hold. The three existing `Sensitive` fields in this provider (`user.password`, and `sso` on the OIDC/SAML app resources) are genuine **secrets**. A logo is not one, and marking it sensitive does propagate to any output referencing it.

The change is right for a different reason — size, which I measured rather than assumed. With a 200KB PNG the logo renders as a single **270,909-character line**, taking the plan from 2.2KB to **273KB**. The API accepts 1MB for a logo and 5MB for a background, so it gets worse in real use. `Sensitive` is the only lever SDKv2 offers to suppress printing, and base64 is not readable by eye anyway.

I checked `StateFunc` hashing first, as the more semantically honest option, and rejected it: with `StateFunc` the SDK hands the transformed value back from `d.Get`, so the resource would upload the hash instead of the image.

Verified the data still reaches the API unchanged — applying a 203,143-byte PNG returns `logo.file_size` 203143, and the follow-up plan is empty. The code comment and the docs both record the size reasoning so it is not later mistaken for a secrecy claim.

**Comment by Subterrane** [comment]
Acted on the suppressed finding from the last pass — it was correct, and chasing it down turned up something bigger.

The inconsistency was real: `ConfiguredKeys` counted `logo = ""` as configured while `RequestBody` dropped it, so an empty string was recorded in state and never sent.

But sending it would not have fixed anything. Probing the API showed `logo` is **three-state**, and `models.Brand` reaches only two of the three:

| request | result |
| --- | --- |
| key omitted | 200, image left alone |
| base64 string | 200, image replaced |
| `null` | 200, **image removed** |
| `""` | 200, **ignored entirely** — image unchanged |
| `false` | 422 |

`models.Brand.Logo` is `*string` tagged `omitempty`, so a nil pointer is *omitted* rather than sent as `null`. **The SDK cannot remove an image at all** — the same structural gap `App` solved with the `ClearPolicyID`/`ClearBrandID` flags.

So the fix is two-sided, in `4aa028a`:

- `RequestBody` now sends whatever `ConfiguredKeys` counted as configured, so the two agree.
- A `ValidateFunc` refuses `""` during `plan`, naming the limitation and saying to omit the argument instead — an error beats a request that quietly changes nothing.

Documented under "Removing an image is not supported", with tests for the validator and for `RequestBody` not second-guessing `ConfiguredKeys`.

Worth flagging for the SDK rather than this PR: giving `Brand` `ClearLogo`/`ClearBackground` flags, exactly as `App` has, would make removal expressible and let this resource support it. Not doing it here — it is an SDK change and this PR should not grow one.

**Comment by Subterrane** [comment]
Acted on the suppressed finding from the last pass — it was right, and it was the most serious one so far.

`brandDelete` called `DeleteBrand` for any brand in state, while the resource documented that the master brand cannot be destroyed here. The master brand is reachable only by importing it, which somebody does to change a colour, and a later `terraform destroy` would have attempted the delete.

**I did not test what the endpoint does with it.** Finding out would have meant deleting the master brand of a shared Development tenant, which is the branding every app falls back to. The answer does not change what this resource should do:

- if the API refuses, the guard is a clearer error than the one it returns
- if the API allows, the guard is the difference between a destroy and an outage

So `3453acb` refuses it with an error naming the brand and pointing at `terraform state rm`.

One thing worth noting beyond the report: the guard had to move **above** the `m.(*onelogin.OneloginSDK)` assertion. Refusing needs no client, and putting it first is what makes the test meaningful — `brandDelete` is called with a nil client, so a guard that let the master brand through panics rather than quietly passing. There is a control test for an ordinary brand too, so the guard cannot degrade into refusing everything.

Mutation-checked: deleting the guard fails the test.
